File: java/parser/src/main/java/gov/lbl/parser/parser/Algorithm_sectionVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Statement;
import gov.lbl.parser.domain.Algorithm_section;

import static java.util.stream.Collectors.toList;

public class Algorithm_sectionVisitor extends modelicaBaseVisitor<Algorithm_section> {
    @Override
    public Algorithm_section visitAlgorithm_section(modelicaParser.Algorithm_sectionContext ctx) {
        Boolean initial = ctx.INITIAL() == null ? false : true;

        StatementVisitor statementVisitor = new StatementVisitor();
        List<Statement> statements = ctx.statement() == null ? null : ctx.statement()
                                                                    .stream()
                                                                    .map(statement -> statement.accept(statementVisitor))
                                                                    .collect(toList());
        return new Algorithm_section(initial, statements);
    }
}

File: jsParser/parser/Algorithm_sectionVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Algorithm_section = require('../domain/Algorithm_section')

const StatementVisitor = require('./StatementVisitor')

class Algorithm_sectionVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitAlgorithm_section (ctx) {
    const initial = !!ctx.INITIAL()
    const statements = []

    if (ctx.statement()) {
      const statementVisitor = new StatementVisitor.StatementVisitor()
      ctx.statement().forEach(stmt => {
        statements.push(statementVisitor.visitStatement(stmt))
      })
    }

    return new Algorithm_section.Algorithm_section(initial, statements)
  }
};

Algorithm_sectionVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitAlgorithm_section = this.visitAlgorithm_section
exports.Algorithm_sectionVisitor = Algorithm_sectionVisitor
